doc/source/user_guide/sparse.rst

```diff
@@ -105,84 +130,117 @@ keeps an arrays of all of the locations where the data are not equal to the
+
+.. code-block:: python
+
+   >>> import warnings
+
+   >>> warnings.filterwarnings('ignore', 'Sparse', FutureWarning)
+   >>> pd.SparseSeries()  # No warning message
+   Series([], dtype: Sparse[float64, nan])
+   BlockIndex
+   Block locations: array([], dtype=int32)
+   Block lengths: array([], dtype=int32)
+
+But we recommend modifying your code, rather than ignoring the warning.
+
+**Construction**
 
-You can change the dtype using ``.astype()``, the result is also sparse. Note that
-``.astype()`` also affects to the ``fill_value`` to keep its dense representation.
+From an array-like, use the regular :class:`Series` or
+:class:`DataFrame` constructors with :class:`SparseArray` values.
 
+.. code-block:: python
+
+   # Old way
+   >>> pd.SparseDataFrame({"A": [0, 1]})
 
 .. ipython:: python
 
-   s = pd.Series([1, 0, 0, 0, 0])
-   s
-   ss = s.to_sparse()
-   ss
-   ss.astype(np.float64)
+   # New way
+   pd.DataFrame({"A": pd.SparseArray([0, 1])})
 
-It raises if any value cannot be coerced to specified dtype.
+From a SciPy sparse matrix, use :meth:`DataFrame.sparse.from_spmatrix`,
 
-.. code-block:: ipython
+.. code-block:: python
 
-   In [1]: ss = pd.Series([1, np.nan, np.nan]).to_sparse()
-   Out[1]:
-   0    1.0
-   1    NaN
-   2    NaN
-   dtype: float64
-   BlockIndex
-   Block locations: array([0], dtype=int32)
-   Block lengths: array([1], dtype=int32)
+   # Old way
+   >>> from scipy import sparse
+   >>> mat = sparse.eye(3)
+   >>> df = pd.SparseDataFrame(mat, columns=['A', 'B', 'C'])
 
-   In [2]: ss.astype(np.int64)
-   Out[2]:
-   ValueError: unable to coerce current fill_value nan to int64 dtype
+.. ipython:: python
 
-.. _sparse.calculation:
+   # New way
+   from scipy import sparse
+   mat = sparse.eye(3)
+   df = pd.DataFrame.sparse.from_spmatrix(mat, columns=['A', 'B', 'C'])
+   df
 
-Sparse Calculation
-------------------
+**Conversion**
 
-You can apply NumPy *ufuncs* to ``SparseArray`` and get a ``SparseArray`` as a result.
+From sparse to dense, use the ``.sparse`` accessors
 
 .. ipython:: python
 
-   arr = pd.SparseArray([1., np.nan, np.nan, -2., np.nan])
-   np.abs(arr)
+   df.sparse.to_dense()
+   df.sparse.to_coo()
+   df['A']
 
+From dense to sparse, use :meth:`DataFrame.astype` with a :class:`SparseDtype`.
 
-The *ufunc* is also applied to ``fill_value``. This is needed to get
-the correct dense result.
+.. ipython:: python
+
+   dense = pd.DataFrame({"A": [1, 0, 0, 1]})
+   dtype = pd.SparseDtype(int, fill_value=0)
+   dense.astype(dtype)
+
+**Sparse Properties**
+
+Sparse-specific properties, like ``density``, are available on the ``.sparse`` accssor.
 
 .. ipython:: python
 
-   arr = pd.SparseArray([1., -1, -1, -2., -1], fill_value=-1)
-   np.abs(arr)
-   np.abs(arr).to_dense()
+   df.sparse.density
+
+The ``SparseDataFrame.default_kind`` and ``SparseDataFrame.default_fill_value`` attributes
+have no replacement.
 
 .. _sparse.scipysparse:
 
```

Comment:
Ideally, I would convert this "Interaction with scipy.sparse" section also to the new way?

Reply:
In addition, it is a bit strange that the "SparseDataFrame" and "SparseSeries" sections (that actually document those classes, and which you reference above) is under the "Interaction with scipy.sparse" one. I would give them their own section? (eg "Deprecated SparseDataFrame and SparseSeries" ?)